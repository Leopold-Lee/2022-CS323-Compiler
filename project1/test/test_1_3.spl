int test_1_3(int a, int b)
{
  int c = a && b || c;
  c = a == b;
  if (a > b)
  {
    return a;
  }
  else if (b == a){
    return b;
  } else {
    return b;
  }
}
