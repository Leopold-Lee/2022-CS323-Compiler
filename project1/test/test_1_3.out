Program (1)
  ExtDefList (1)
    ExtDef (1)
      Specifier (1)
        TYPE: int
      FunDec (1)
        ID: test_1_3
        LP
        VarList (1)
          ParamDec (1)
            Specifier (1)
              TYPE: int
            VarDec (1)
              ID: a
          COMMA
          VarList (1)
            ParamDec (1)
              Specifier (1)
                TYPE: int
              VarDec (1)
                ID: b
        RP
      CompSt (2)
        LC
        StmtList (3)
          Stmt (3)
            IF
            LP
            Exp (3)
              Exp (3)
                ID: a
              GT
              Exp (3)
                ID: b
            RP
            Stmt (4)
              CompSt (4)
                LC
                StmtList (5)
                  Stmt (5)
                    RETURN
                    Exp (5)
                      ID: a
                    SEMI
                RC
            ELSE
            Stmt (7)
              IF
              LP
              Exp (7)
                Exp (7)
                  ID: b
                EQ
                Exp (7)
                  ID: a
              RP
              Stmt (7)
                CompSt (7)
                  LC
                  StmtList (8)
                    Stmt (8)
                      RETURN
                      Exp (8)
                        ID: b
                      SEMI
                  RC
              ELSE
              Stmt (9)
                CompSt (9)
                  LC
                  StmtList (10)
                    Stmt (10)
                      RETURN
                      Exp (10)
                        ID: b
                      SEMI
                  RC
        RC
